// @trezor/transport/lib/transports/udp.browser.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UdpTransport = void 0;
const abstract_1 = require("./abstract");
const result_1 = require("../utils/result");
const errors_1 = require("../errors");
const emptyAbortable = () => ({
    promise: Promise.resolve((0, result_1.error)({ error: errors_1.WRONG_ENVIRONMENT })),
    abort: () => { },
});
const empty = () => Promise.resolve((0, result_1.error)({ error: errors_1.WRONG_ENVIRONMENT }));
const emptySync = () => (0, result_1.error)({ error: errors_1.WRONG_ENVIRONMENT });
class UdpTransport extends abstract_1.AbstractTransport {
    constructor() {
        super(...arguments);
        this.name = 'UdpTransport';
        this.init = emptyAbortable;
        this.acquire = emptyAbortable;
        this.enumerate = emptyAbortable;
        this.call = emptyAbortable;
        this.receive = emptyAbortable;
        this.send = emptyAbortable;
        this.release = emptyAbortable;
        this.stop = empty;
        this.releaseDevice = empty;
        this.listen = emptySync;
    }
}
exports.UdpTransport = UdpTransport;
