// @trezor/connect/lib/backend/BackendManager.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackendManager = void 0;
const DataManager_1 = require("../data/DataManager");
const constants_1 = require("../constants");
const Blockchain_1 = require("./Blockchain");
const events_1 = require("../events");
class BackendManager {
    constructor() {
        this.instances = {};
        this.custom = {};
        this.preferred = {};
        this.reconnect = {};
    }
    get(shortcut) {
        var _a;
        return (_a = this.instances[shortcut]) !== null && _a !== void 0 ? _a : null;
    }
    async getOrConnect(coinInfo, postMessage) {
        let backend = this.instances[coinInfo.shortcut];
        if (!backend) {
            backend = new Blockchain_1.Blockchain({
                coinInfo: this.patchCoinInfo(coinInfo),
                postMessage,
                debug: DataManager_1.DataManager.getSettings('debug'),
                proxy: DataManager_1.DataManager.getSettings('proxy'),
                onDisconnected: pendingSubscriptions => {
                    const reconnectAttempts = pendingSubscriptions ? 0 : undefined;
                    this.onDisconnect(coinInfo, postMessage, reconnectAttempts);
                },
            });
            this.setInstance(coinInfo.shortcut, backend);
        }
        const reconnect = this.clearReconnect(coinInfo.shortcut);
        try {
            const info = await backend.init();
            this.setPreferred(coinInfo.shortcut, info.url);
            return backend;
        }
        catch (error) {
            this.onDisconnect(coinInfo, postMessage, reconnect === null || reconnect === void 0 ? void 0 : reconnect.attempts);
            throw error;
        }
    }
    dispose() {
        Object.keys(this.reconnect).forEach(this.clearReconnect, this);
        Object.values(this.instances).forEach(i => i.disconnect());
    }
    reconnectAll() {
        Object.keys(this.reconnect).forEach(this.clearReconnect, this);
        const params = Object.values(this.instances).map(i => [i.coinInfo, i.postMessage]);
        Object.values(this.instances).forEach(i => i.disconnect());
        return Promise.all(params.map(p => this.getOrConnect(...p)));
    }
    isSupported(coinInfo) {
        const info = this.custom[coinInfo.shortcut] || coinInfo.blockchainLink;
        if (!info) {
            throw constants_1.ERRORS.TypedError('Backend_NotSupported');
        }
    }
    setCustom(shortcut, blockchainLink) {
        this.setPreferred(shortcut, undefined);
        this.custom[shortcut] = blockchainLink;
    }
    removeCustom(shortcut) {
        this.setPreferred(shortcut, undefined);
        delete this.custom[shortcut];
    }
    setInstance(shortcut, instance) {
        if (!instance)
            delete this.instances[shortcut];
        else
            this.instances[shortcut] = instance;
    }
    setPreferred(shortcut, url) {
        if (!url)
            delete this.preferred[shortcut];
        else
            this.preferred[shortcut] = url;
    }
    onDisconnect(coinInfo, postMessage, reconnectAttempt) {
        var _a;
        this.setInstance(coinInfo.shortcut, undefined);
        if (reconnectAttempt === undefined || reconnectAttempt === 4) {
            this.setPreferred(coinInfo.shortcut, undefined);
        }
        if (reconnectAttempt === undefined) {
            return;
        }
        const timeout = Math.min(2500 * reconnectAttempt, 20000);
        const time = Date.now() + timeout;
        const handle = setTimeout(() => {
            this.getOrConnect(coinInfo, postMessage).catch(() => { });
        }, timeout);
        clearTimeout((_a = this.reconnect[coinInfo.shortcut]) === null || _a === void 0 ? void 0 : _a.handle);
        this.reconnect[coinInfo.shortcut] = { attempts: reconnectAttempt + 1, handle };
        postMessage((0, events_1.createBlockchainMessage)(events_1.BLOCKCHAIN.RECONNECTING, { coin: coinInfo, time }));
    }
    clearReconnect(shortcut) {
        const reconnect = this.reconnect[shortcut];
        clearTimeout(reconnect === null || reconnect === void 0 ? void 0 : reconnect.handle);
        delete this.reconnect[shortcut];
        return reconnect;
    }
    patchCoinInfo(coinInfo) {
        var _a, _b;
        const custom = this.custom[coinInfo.shortcut];
        const preferred = this.preferred[coinInfo.shortcut];
        const url = preferred ? [preferred] : (_a = custom === null || custom === void 0 ? void 0 : custom.url) !== null && _a !== void 0 ? _a : (_b = coinInfo.blockchainLink) === null || _b === void 0 ? void 0 : _b.url;
        return {
            ...coinInfo,
            blockchainLink: {
                ...coinInfo.blockchainLink,
                ...custom,
                url,
            },
        };
    }
}
exports.BackendManager = BackendManager;
