// @trezor/utxo-lib/lib/script/scriptSignature.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.encode = exports.decode = exports.fromDER = exports.toDER = void 0;
const tslib_1 = require("tslib");
const bip66_1 = tslib_1.__importDefault(require("bip66"));
const types = tslib_1.__importStar(require("../types"));
const ZERO = Buffer.alloc(1, 0);
function toDER(x) {
    let i = 0;
    while (x[i] === 0)
        ++i;
    if (i === x.length)
        return ZERO;
    x = x.subarray(i);
    if (x[0] & 0x80)
        return Buffer.concat([ZERO, x], 1 + x.length);
    return x;
}
exports.toDER = toDER;
function fromDER(x) {
    if (x[0] === 0x00)
        x = x.subarray(1);
    const buffer = Buffer.alloc(32, 0);
    const bstart = Math.max(0, 32 - x.length);
    x.copy(buffer, bstart);
    return buffer;
}
exports.fromDER = fromDER;
function decode(buffer) {
    const hashType = buffer.readUInt8(buffer.length - 1);
    const hashTypeMod = hashType & ~0x80;
    if (hashTypeMod <= 0 || hashTypeMod >= 4)
        throw new Error(`Invalid hashType ${hashType}`);
    const decoded = bip66_1.default.decode(buffer.subarray(0, -1));
    const r = fromDER(decoded.r);
    const s = fromDER(decoded.s);
    const signature = Buffer.concat([r, s], 64);
    return { signature, hashType };
}
exports.decode = decode;
function encode(signature, hashType) {
    types.typeforce({
        signature: types.BufferN(64),
        hashType: types.UInt8,
    }, { signature, hashType });
    const hashTypeMod = hashType & ~0x80;
    if (hashTypeMod <= 0 || hashTypeMod >= 4)
        throw new Error(`Invalid hashType ${hashType}`);
    const hashTypeBuffer = Buffer.allocUnsafe(1);
    hashTypeBuffer.writeUInt8(hashType, 0);
    const r = toDER(signature.subarray(0, 32));
    const s = toDER(signature.subarray(32, 64));
    return Buffer.concat([bip66_1.default.encode(r, s), hashTypeBuffer]);
}
exports.encode = encode;
