// @trezor/transport/lib/utils/bridgeApiCall.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.bridgeApiCall = void 0;
const tslib_1 = require("tslib");
const result_1 = require("./result");
const ERRORS = tslib_1.__importStar(require("../errors"));
const protocol_1 = require("@trezor/protocol");
const applyContentTypeHeader_1 = require("./applyContentTypeHeader");
const IS_NODE = typeof process !== 'undefined' && typeof window === 'undefined';
function contentType(body) {
    if (typeof body === 'string') {
        if (body === '') {
            return 'text/plain';
        }
        return 'application/octet-stream';
    }
    return 'application/json';
}
function wrapBody(body) {
    if (typeof body === 'string') {
        return body;
    }
    return JSON.stringify(body);
}
function parseResult(text) {
    try {
        return JSON.parse(text);
    }
    catch (e) {
        return text;
    }
}
function bridgeApiCall(options) {
    return tslib_1.__awaiter(this, void 0, void 0, function* () {
        const fetchOptions = {
            method: options.method,
            body: wrapBody(options.body),
            credentials: 'same-origin',
            headers: {},
            signal: options.signal,
            timeout: options.timeout,
        };
        fetchOptions.headers = (0, applyContentTypeHeader_1.applyContentTypeHeader)({
            headers: fetchOptions.headers,
            contentType: contentType(options.body == null ? '' : options.body),
            skipContentTypeHeader: options.skipContentTypeHeader,
        });
        if (IS_NODE) {
            fetchOptions.headers = Object.assign(Object.assign({}, fetchOptions.headers), { Origin: 'https://node.trezor.io' });
        }
        let res;
        try {
            res = yield fetch(options.url, fetchOptions);
        }
        catch (err) {
            return (0, result_1.error)({ error: ERRORS.HTTP_ERROR, message: err.message });
        }
        let resParsed;
        try {
            resParsed = yield res.text();
            resParsed = parseResult(resParsed);
        }
        catch (err) {
            return (0, result_1.error)({ error: ERRORS.HTTP_ERROR, message: err.message });
        }
        const BRIDGE_ERROR_DEVICE_CLOSED = 'closed device';
        const BRIDGE_MALFORMED_PROTOBUF = 'malformed protobuf';
        const BRIDGE_MALFORMED_WIRE_FORMAT = 'malformed wire format';
        if (!res.ok) {
            const errStr = typeof resParsed !== 'string' && 'error' in resParsed
                ? resParsed.error
                : resParsed;
            if (errStr === BRIDGE_ERROR_DEVICE_CLOSED) {
                return (0, result_1.error)({ error: ERRORS.INTERFACE_UNABLE_TO_OPEN_DEVICE });
            }
            if (errStr === BRIDGE_MALFORMED_PROTOBUF) {
                return (0, result_1.error)({ error: protocol_1.PROTOCOL_MALFORMED });
            }
            if (errStr === BRIDGE_MALFORMED_WIRE_FORMAT) {
                return (0, result_1.error)({ error: protocol_1.PROTOCOL_MALFORMED });
            }
            return (0, result_1.unknownError)(new Error(errStr), [
                ERRORS.DEVICE_NOT_FOUND,
                ERRORS.HTTP_ERROR,
                ERRORS.DEVICE_DISCONNECTED_DURING_ACTION,
                ERRORS.OTHER_CALL_IN_PROGRESS,
                ERRORS.SESSION_NOT_FOUND,
                ERRORS.SESSION_WRONG_PREVIOUS,
            ]);
        }
        return (0, result_1.success)(resParsed);
    });
}
exports.bridgeApiCall = bridgeApiCall;
