// @trezor/transport/lib/utils/bridgeApiResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.empty = exports.call = exports.acquire = exports.devices = exports.version = exports.info = void 0;
const tslib_1 = require("tslib");
const result_1 = require("./result");
const ERRORS = tslib_1.__importStar(require("../errors"));
function isString(payload) {
    return typeof payload === 'string';
}
function info(res) {
    if (isString(res)) {
        return (0, result_1.error)({ error: ERRORS.WRONG_RESULT_TYPE });
    }
    const { version } = res;
    if (typeof version !== 'string') {
        return (0, result_1.error)({ error: ERRORS.WRONG_RESULT_TYPE });
    }
    const configured = !!res.configured;
    return (0, result_1.success)({ version, configured });
}
exports.info = info;
function version(res) {
    if (!isString(res)) {
        return (0, result_1.error)({ error: ERRORS.WRONG_RESULT_TYPE });
    }
    return (0, result_1.success)(res.trim());
}
exports.version = version;
function devices(res) {
    if (isString(res)) {
        return (0, result_1.error)({ error: ERRORS.WRONG_RESULT_TYPE });
    }
    if (!(res instanceof Array)) {
        return (0, result_1.error)({ error: ERRORS.WRONG_RESULT_TYPE });
    }
    if (res.some(o => typeof o !== 'object' ||
        !o ||
        typeof o.path !== 'string' ||
        (typeof o.session !== 'string' && o.session !== null))) {
        return (0, result_1.error)({ error: ERRORS.WRONG_RESULT_TYPE });
    }
    return (0, result_1.success)(res.map((o) => ({
        path: o.path,
        session: o.session,
        product: o.product,
        vendor: o.vendor,
        debug: o.debug,
        debugSession: o.debugSession,
    })));
}
exports.devices = devices;
function acquire(res) {
    if (isString(res)) {
        return (0, result_1.error)({ error: ERRORS.WRONG_RESULT_TYPE });
    }
    const { session } = res;
    if (typeof session !== 'string') {
        return (0, result_1.error)({ error: ERRORS.WRONG_RESULT_TYPE });
    }
    return (0, result_1.success)(session);
}
exports.acquire = acquire;
function call(res) {
    if (!isString(res)) {
        return (0, result_1.error)({ error: ERRORS.WRONG_RESULT_TYPE });
    }
    return (0, result_1.success)(res);
}
exports.call = call;
function empty(res) {
    return res != null && JSON.stringify(res) === '{}'
        ? (0, result_1.error)({ error: ERRORS.WRONG_RESULT_TYPE })
        : (0, result_1.success)(undefined);
}
exports.empty = empty;
