// @trezor/utxo-lib/lib/crypto.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.hmacSHA512 = exports.hash256 = exports.hash160blake256 = exports.hash160 = exports.blake256 = exports.sha256 = exports.sha1 = exports.ripemd160 = void 0;
const tslib_1 = require("tslib");
const blake_hash_1 = tslib_1.__importDefault(require("blake-hash"));
const create_hash_1 = tslib_1.__importDefault(require("create-hash"));
const create_hmac_1 = tslib_1.__importDefault(require("create-hmac"));
function ripemd160(buffer) {
    try {
        return (0, create_hash_1.default)('rmd160').update(buffer).digest();
    }
    catch (err) {
        return (0, create_hash_1.default)('ripemd160').update(buffer).digest();
    }
}
exports.ripemd160 = ripemd160;
function sha1(buffer) {
    return (0, create_hash_1.default)('sha1').update(buffer).digest();
}
exports.sha1 = sha1;
function sha256(buffer) {
    return (0, create_hash_1.default)('sha256').update(buffer).digest();
}
exports.sha256 = sha256;
function blake256(buffer) {
    return (0, blake_hash_1.default)('blake256').update(buffer).digest();
}
exports.blake256 = blake256;
function hash160(buffer) {
    return ripemd160(sha256(buffer));
}
exports.hash160 = hash160;
function hash160blake256(buffer) {
    return ripemd160(blake256(buffer));
}
exports.hash160blake256 = hash160blake256;
function hash256(buffer) {
    return sha256(sha256(buffer));
}
exports.hash256 = hash256;
function hmacSHA512(key, data) {
    return (0, create_hmac_1.default)('sha512', key).update(data).digest();
}
exports.hmacSHA512 = hmacSHA512;
